### PR TITLE
feat: add agent annotations @AgentTool, @AgentRisk, @AgentInternal, @AgentExclude, @AgentToolParam

### PR DESCRIPTION
**Pull request by arrrrny**
## Summary\n\nCloses #114\n\nAdds agent-centric architecture annotations for zuraffa AgentPlugin integration. Annotations compile into a typed const AgentDirective in generated .zorphy.dart files, consumable by AgentPlugin AST scan.\n\n### New Annotations\n\n- @AgentTool(name?, namespace?, description?) - Override generated MCP tool identity\n- @AgentRisk(AgentRiskTier) - Declare risk tier: safe, confirm, or admin\n- @AgentInternal - Sugar for @AgentRisk(admin) + excluded from public manifests\n- @AgentExclude - Entity/usecase never becomes a tool\n- @AgentToolParam(description?) - Field-level tool parameter description\n\n### Contract Types (in zorphy_annotation)\n\n- AgentRiskTier enum (safe, confirm, admin)\n- AgentDirective - top-level const emitted in generated files\n- AgentToolDirective - tool identity data\n- AgentParamDirective - per-parameter metadata\n\n### Test plan\n\n- [x] dart analyze - no new errors or warnings\n- [x] dart test - 8/8 new tests pass, 206/213 total (7 pre-existing fixture failures)\n- [ ] Integration test in zuraffa (AgentPlugin consumes emitted directive)

### CODE REVIEW

**Comment by arrrrny** [comment]
Fetch URLs
2 pages

Looking at the diff and PR description, this adds agent annotations that compile into AgentDirective constants in generated .zorphy.dart files for consumption by zuraffa's AgentPlugin.
Findings:
_escape in agent_directive_generator.dart is incomplete — it only escapes single quotes ('), but not backslashes (\) or dollar signs ($). This produces broken generated code when annotation strings contain newlines (\n), literal backslashes (\\), or dollar signs ($name). A string like description: 'path\to\file' becomes invalid Dart ( \t → tab, \f → form feed), and $var triggers string interpolation in the generated file where var likely doesn't exist, causing compile errors in the generated .zorphy.dart.
AgentDirectiveGenerator emits final agentDirective instead of const agentDirective — the class docstring explicitly promises "a top-level const agentDirective = AgentDirective(...)" and the PR says this const is "consumed by zuraffa's AgentPlugin via AST scan". A final variable with a const initializer is not a const declaration (isConst == false in analyzer terms); AST scanners looking for const top-level variables will miss it.
Missing tests for _escape edge cases — no test covers strings with quotes, backslashes, or dollar signs.
Missing test verifying the emitted declaration is const — no test checks that the output uses const agentDirective rather than final agentDirective.
diff
Copy
diff --git a/zorphy/lib/src/generators/agent_directive_generator.dart b/zorphy/lib/src/generators/agent_directive_generator.dart
index 00b4d40..e3f8a2a 100644
--- a/zorphy/lib/src/generators/agent_directive_generator.dart
+++ b/zorphy/lib/src/generators/agent_directive_generator.dart
@@ -30,7 +30,7 @@ class AgentDirectiveGenerator extends UniversalGenerator {
   @override
   bool shouldGenerate(GenerationContext context) {
     return context.metadata.agentDirectiveInfo.hasAgentAnnotations;
   }
 
   /// Builds the AgentDirective constructor argument string.
@@ -75,8 +75,9 @@ class AgentDirectiveGenerator extends UniversalGenerator {
     if (parts.isEmpty) {
       return 'const AgentDirective()';
     }
     return 'const AgentDirective(\n    ${parts.join('\n    ')}\n  )';
   }
 
-  /// Escapes single quotes in a string.
-  String _escape(String s) => s.replaceAll("'", "\\'");
+  /// Escapes single quotes, backslashes and dollar signs in a string
+  /// so it can be safely embedded in a single-quoted Dart string literal.
+  String _escape(String s) => s.replaceAll('\\', '\\\\').replaceAll("'", "\\'").replaceAll('\$', '\\\$');
 }
diff --git a/zorphy/lib/src/generators/agent_directive_generator.dart b/zorphy/lib/src/generators/agent_directive_generator.dart
index e3f8a2a..c5e8d1b 100644
--- a/zorphy/lib/src/generators/agent_directive_generator.dart
+++ b/zorphy/lib/src/generators/agent_directive_generator.dart
@@ -18,7 +18,7 @@ class AgentDirectiveGenerator extends UniversalGenerator {
 
     final fields = _buildFields(info);
     final directive = Field((f) {
       f.name = 'agentDirective';
-      f.modifier = FieldModifier.final$;
+      f.modifier = FieldModifier.constant;
       f.type = referType('AgentDirective');
       f.assignment = Code(fields);
     });
diff --git a/zorphy/test/generation/issue_114_agent_annotations_test.dart b/zorphy/test/generation/issue_114_agent_annotations_test.dart
index 016c65a..8f9e2b1 100644
--- a/zorphy/test/generation/issue_114_agent_annotations_test.dart
+++ b/zorphy/test/generation/issue_114_agent_annotations_test.dart
@@ -170,6 +170,35 @@ void main() {
       expect(code, contains('AgentParamDirective'));
       expect(code, contains("'query'"));
       expect(code, contains("'limit'"));
+    });
+
+    test('escapes special characters in strings', () {
+      final meta = _concreteMeta(
+        name: 'EscapeTest',
+        agentDirectiveInfo: const AgentDirectiveInfo(
+          toolName: "it's",
+          toolDescription: 'path\\to\\file',
+          paramDescriptions: {'\$var': 'costs \$5'},
+          hasAgentAnnotations: true,
+        ),
+      );
+      final context = GenerationContext(metadata: meta, config: _bareConfig());
+      final specs = AgentDirectiveGenerator().generateSpec(context);
+      final code = _emitSpecs(specs);
+      expect(code, contains("name: 'it\\'s'"));
+      expect(code, contains("description: 'path\\\\to\\\\file'"));
+      expect(code, contains("'\\$var': const AgentParamDirective(description: 'costs \\\$5')"));
+    });
+
+    test('emits const declaration, not final', () {
+      final meta = _concreteMeta(
+        name: 'Simple',
+        agentDirectiveInfo: const AgentDirectiveInfo(
+          hasAgentAnnotations: true,
+        ),
+      );
+      final context = GenerationContext(metadata: meta, config: _bareConfig());
+      final specs = AgentDirectiveGenerator().generateSpec(context);
+      final code = _emitSpecs(specs);
+      expect(code, contains('const agentDirective'));
+      expect(code, isNot(contains('final agentDirective')));
     });
 
     test('safe risk tier is omitted from output (default)', () {
local review verdict: findings — (1) _escape omits backslash and $ escaping, producing invalid generated Dart; (2) AgentDirectiveGenerator emits final instead of promised const, breaking AgentPlugin AST-scan contract; (3-4) missing tests for escape edge cases and const modifier.
